server/seeder.js

```diff
@@ -1,60 +1,122 @@
+const mongoose = require('mongoose');
 const Listing = require('./models/Listing');
 
-
-const seeder = () => {
-
-  // Droping existing sample data
-  Listing.deleteMany({}, () => {
-
-
-    for(let j = 1; j <= 100; j++){
-
-      let details = [];
-
-      for(let i = 1; i <= 50; i++){
-
-        let d = faker.date.between('2018-01-01', '2019-09-30');
-        const newD = moment(d).startOf('day')
-        
-        detail = {
-          date: newD,
-          guests: {
-            adults: faker.random.number({'min': 1, 'max': 3}),
-            children: faker.random.number({'min': 0, 'max': 3}),
-            infants: faker.random.number({'min': 0, 'max': 3}),
-          }
+mongoose.connect(process.env.MONGODB_URI || 'mongodb://localhost/booking', {
+  useNewUrlParser: true
+});
+let db = mongoose.connection;
+
+db.on('error', (err) => {
+  console.log('error connecting', err);
+});
+
+db.once('open', () => {
+  console.log('Connected to DB');
+});
+
+let recordId = 0;
+let insertData = [];
+const useModel = false;
+const batches = 100;
+const recordsPerBatch = 100000;
+let progressInserted = 0;
+
+const seeder = async () => {
+  let batchCounter = 0;
+  while (batchCounter < batches) {
+    if (progressInserted === 0) {
+      console.log(`Starting to insert ${recordsPerBatch * batches} records. Standby...`)
+    } else {
+      console.log(`${progressInserted} of ${recordsPerBatch * batches} inserted. Standby...`)
+    }
+    generateBatch();
+    await insert(useModel);
+    progressInserted += recordsPerBatch;
+    insertData = [];
+    batchCounter++;
+  }
+
+  if (progressInserted === (recordsPerBatch * batches)) {
+    console.log('Done!');
+  }
+};
+
+const insert = (useModel) => {
+  if (useModel) {
+    return new Promise(function (resolve, reject) {
+      Listing.insertMany(insertData, function (error, doc) {
+        if (error) {
+          console.log(error);
+        } else {
+          resolve();
         }
-        details.push(detail)
-      }
-
-      const newListing = new Listing({
-
-        listing_id: j,
-
-        details: details,
-
-        listing_price: faker.commerce.price(50, 100)
-      }) 
-
-     newListing.save((err) => {
-        if(err){
-          console.log(err)
+      });
+    });
+  } else {
+    return new Promise(function (resolve, reject) {
+      db.collection('listing').insertMany(insertData, function (error, doc) {
+        if (error) {
+          console.log(error);
+        } else {
+          resolve();
         }
-      })
-    }
- })
+      });
+    });
+  }
 }
 
-// Call Seeder func
-seeder();
-
-
+const generateBatch = () => {
+
+  let recordsCounter = 1;
+  let batch = [];
+
+  while (recordsCounter <= recordsPerBatch) {
+    let details = [];
+    let bookingsCounter = 1;
+
+    while (bookingsCounter <= 1) {
+      let d = faker.date.between('2018-01-01', '2019-09-30');
+      const newD = moment(d).startOf('day');
+      detail = {
+        date: newD,
```

Comment:
replace `newD` with `d` and the I believe the date will render properly

Reply:
done